lib/storage.js

```diff
@@ -79,7 +99,48 @@ async function putRendition(rendition) {
     }
 }
 
+async function getWatermark(params, inDirectory, disableDownload) {
+    // Note: validation has happened in validate.js before
+    if (process.env.WORKER_TEST_MODE) {
+        // local file support for `Asset Compute test-worker` unit tests
+        // not supported for clients in production
+        // params.watermarkContent will just be a filename like 'watermark.png'
+        const filePath = path.join(inDirectory, params.watermarkContent);
+        if (!fs.existsSync(filePath)) {
+            throw new Error(`Invalid or missing local file ${params.watermarkContent}`);
+        }
+        return new Watermark(params, inDirectory);
+
+    } else {
+        const watermark = new Watermark(params, inDirectory);
+
+        if (path.extname(watermark.name).toLowerCase() !== ".png" && !watermark.url.startsWith("data:image/png")) {
+            throw new RenditionFormatUnsupportedError(`Invalid watermark format ${watermark.url}`);
+        }
+
+        if (disableDownload) {
+            console.log(`Skipping watermark file download for ${AssetComputeLogUtils.redactUrl(params.watermarkContent)}`);
```

Comment:
@DrGirlfriend I think we dont need `disableDownload` for watermark. we can remove this condition ? 
Ex: This causes worker-ffmpeg(which sets `disableSourceDownload=true` ) watermarking to fail.